op-challenger/game/fault/trace/asterisc/provider.go

```diff
@@ -0,0 +1,224 @@
+package asterisc
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"path/filepath"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/config"
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/trace/utils"
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
+	"github.com/ethereum-optimism/optimism/op-program/host/kvstore"
+	"github.com/ethereum-optimism/optimism/op-service/ioutil"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+const (
+	proofsDir      = "proofs"
+	diskStateCache = "state.json.gz"
+)
+
+type AsteriscMetricer interface {
+	RecordAsteriscExecutionTime(t float64)
+}
+
+type AsteriscTraceProvider struct {
+	logger         log.Logger
+	dir            string
+	prestate       string
+	generator      utils.ProofGenerator
+	gameDepth      types.Depth
+	preimageLoader *utils.PreimageLoader
+
+	// lastStep stores the last step in the actual trace if known. 0 indicates unknown.
+	// Cached as an optimisation to avoid repeatedly attempting to execute beyond the end of the trace.
+	lastStep uint64
+}
+
+func NewTraceProvider(logger log.Logger, m AsteriscMetricer, cfg *config.Config, localInputs utils.LocalGameInputs, dir string, gameDepth types.Depth) *AsteriscTraceProvider {
+	return &AsteriscTraceProvider{
+		logger:         logger,
+		dir:            dir,
+		prestate:       cfg.AsteriscAbsolutePreState,
+		generator:      NewExecutor(logger, m, cfg, localInputs),
+		gameDepth:      gameDepth,
+		preimageLoader: utils.NewPreimageLoader(kvstore.NewDiskKV(utils.PreimageDir(dir)).Get),
+	}
+}
+
+func (p *AsteriscTraceProvider) SetMaxDepth(gameDepth types.Depth) {
+	p.gameDepth = gameDepth
+}
```

Comment:
This seems to be unused and can be removed.  Feel free to remove it from the cannon provider while you're at it. :)